src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -234,34 +233,14 @@ async Task ProcessInitialEvaluationDataAsync(CancellationToken cancellationToken
 
                 // Call into Roslyn to initialize language service for this project
                 _context = await _workspaceProjectContextFactory.Value.CreateProjectContextAsync(
-                    languageName,
-                    _contextId,
-                    projectFilePath,
                     _projectGuid,
+                    _contextId,
+                    languageName,
+                    evaluationData,
                     hostObject,
-                    binOutputPath,
-                    assemblyName,
                     cancellationToken);
 
                 _disposableBag.Add(_context);
-
-                // Update additional properties within a batch to avoid thread pool starvation.
-                // https://github.com/dotnet/project-system/issues/8027
-                _context.StartBatch();
-
-                try
-                {
-                    _context.LastDesignTimeBuildSucceeded = false; // By default, turn off diagnostics until the first design time build succeeds for this project.
-
-                    // Pass along any early approximation we have of the command line options
-#pragma warning disable CS0618 // This was obsoleted in favor of the one that takes an array, but here just the string is easier; we'll un-Obsolete this API
-                    _context.SetOptions(commandLineArgsForDesignTimeEvaluation ?? "");
-#pragma warning restore CS0618 // Type or member is obsolete
-                }
-                finally
-                {
-                    await _context.EndBatchAsync();
-                }
             }
             catch (Exception ex)
```

Comment:
>Exception [](http://example.com/codeflow?start=19&length=9)

Roslyn throws `InvalidProjectPropertyValueException` if a property that is required isn't present or has invalid value, in case you want to handle it in a different way.